### PR TITLE
Implement generic logging

### PR DESCRIPTION
**Pull request by velddev**
Adds logging for both Discord and IOstream. A bunch of functions have been moved back and forth to be reusable.

Still need to do a few comments here and there, but I think this is a good start.

Few examples:

Default discord logging
```go
logger := NewDiscordLoggerFromWebhook("test", "<YOUR WEBHOOK URL>")
logger.Warn("CPU temp >80 degrees!")
```

Using decorators:
```go
logger := logger.NewConsoleLoggerBasic()
logger = ApplyLevelBasedLogFilter(Warning, logger)
logger.Message("Ready") // Won't appear.
logger.Warning("Dependencies not properly set up. Please look into this") // Appears
```

Also:
- Fixed SendWebhook not sending properly using easy_embed.go
- Fixed config test not passing

### CODE REVIEW

File: botutil/easyEmbed.go [hunk 1]
```diff
@@ -160,12 +160,20 @@ func (e *Embed) TruncateFields() *Embed {
...
+		Embeds: embedArray,
+	}
+
+	embedJson, err := json.Marshal(params)
 	if err != nil {
 		return err
 	}
-	if _, err = http.Post(Webhook, "application/json", bytes.NewBuffer(embed)); err != nil {
+
+	_, err = http.Post(Webhook, "application/json", bytes.NewBuffer(embedJson))
```
Comment (mandjevant) [inline botutil/easyEmbed.go:173]: Why this separation? Not that it's wrong, but I'm just wondering why

**Review by mandjevant — APPROVED** [review]
Looks good to me, nice function, solid implementation.

File: botutil/easyEmbed.go (hunk 1 above)
Comment (velddev) [reply botutil/easyEmbed.go:173]: Sometimes I needed to fetch the response, and it wouldn't allow me to create a response object inline the statement.

Comment (mandjevant) [reply botutil/easyEmbed.go:173]: Ah, makes sense

**Review by Elian0213 — APPROVED** [review]

**Comment by dhghf** [comment]
I will send my review shortly.

File: logger/decorators/level_filter.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package decorators
+
+import "github.com/Floor-Gang/utilpkg/logger"
+
+type LogLevel int
+
+const (
```
Comment (dhghf) [inline logger/decorators/level_filter.go:7]: A debug level would be helpful.

File: logger/logger.go [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package logger
+
+// Logger is a type that can be used to abstract away logging and add specific
+type Logger interface {
```
Comment (dhghf) [inline logger/logger.go:4]: A debug level would be helpful.

File: logger/discord_logger.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
+package logger
```
Comment (dhghf) [inline logger/discord_logger.go:1]: `go fmt` this file.

File: logger/stream_logger.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package logger
```
Comment (dhghf) [inline logger/stream_logger.go:1]: `go fmt` this file

File: logger/decorators/level_filter.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package decorators
```
Comment (dhghf) [inline logger/decorators/level_filter.go:1]: `go fmt` this file

File: test/config_test.go [hunk 7]
```diff
@@ -5,7 +5,7 @@ import (
 	"strings"
```
Comment (dhghf) [inline test/config_test.go:5]: this should be part of this PR?

**Review by dylhack — COMMENTED** [review]
This PR looks excellent. I think it deserves to be in a separate repository once you believe it's ready for production.

File: test/config_test.go (hunk 7 above)
Comment (velddev) [reply test/config_test.go:5]: Yes, as I mentioned; I fixed the tests so that it would actually pass.

Comment (dhghf) [reply test/config_test.go:5]: gotcha I forgot I moved the config package.

**Review by dylhack — APPROVED** [review]
Looks good. Hopefully, we can get this in it's separate repo.

**Pull request merged**